src/components/DateRangeInput/Presets.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import Box from '../Box';
+
+type OnSelectCallback = (date: [Dayjs, Dayjs]) => void;
+
+interface PresetsProps {
+  currentDateRange: [Dayjs, Dayjs];
+  onSelect: (dates: [Dayjs, Dayjs]) => void;
+}
+
+interface ListItemProps {
+  selected?: boolean;
+  onSelect: (evt: React.KeyboardEvent) => void;
+}
+
+const ListItem: React.FC<ListItemProps> = ({ selected, onSelect, ...rest }) => (
+  <Box
+    as="li"
+    mb={6}
+    cursor="pointer"
+    color="navyblue-100"
+    _selected={{ color: 'gray-50' }}
+    _hover={{ color: 'gray-50' }}
+    aria-selected={selected}
+    onClick={onSelect}
+    fontSize="medium"
+    {...rest}
+  />
+);
+
+const now = dayjs();
+const lastDay = now.subtract(1, 'day');
+const lastWeek = now.subtract(1, 'week');
+const lastMonth = now.subtract(1, 'month');
+const lastThreeMonths = now.subtract(3, 'month');
+const lastSixMonths = now.subtract(6, 'month');
+
+const options = [
```

Comment:
@3nvi Since we need to ensure that each range is visible I am dynamically changing the current month in order to highlight the preset selected.

Reply:
I don't think this is related to what I mention. The "last 24h" should occur from the time   I clicked and **not** from the time the I loaded the tab that  serves the website. 

Am I misunderstanding something in your response?

Reply:
I fixed this one as it can be seen here
https://github.com/panther-labs/pounce/blob/d4ed50ec62262da52ba20666aecef75dca244d0a/src/components/DateRangeInput/Presets.tsx#L42